spec/rubocop/cop/style/instance_vars_in_class_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::InstanceVarsInClass do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  context 'when using instance_variable by itself' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<-RUBY.strip_indent)
+        @instance_variable
+      RUBY
+    end
+  end
+
+  context 'in a class' do
+    context 'when using an instance variable instead of an attr_reader' do
+      it 'registers an offense' do
+        expect_offense(<<-RUBY.strip_indent)
+          class TestClass
+            def foo
+              @instance_variable
+              ^^^^^^^^^^^^^^^^^^ Outside of the initializer, use an `attr_reader :instance_variable` instead of `@instance_variable`.
```

Comment:
In case there's no `attr_*` declared for `@instance_variable`, is it sill fair to enforce this cop?

This also brings up another point which is private instance variables, as one might not have an `attr_*` for them just for internal use.

Reply:
My phrasing probably isn't very good; this is meant to tell users to define an `attr_*` and then use that to interact with the `@instance_variable`.

It's meant to apply in the case of private instance variables as well. Basically this is the goal: https://github.com/rubocop-hq/ruby-style-guide/issues/538#issuecomment-208771509

I wasn't sure how to convey the private part concisely, though, so I left it out. I'll try and come up with a better message.

Reply:
@marcotc Updated the message. It's longer now, unfortunately, but more descriptive.